Modix.Data.Test/Repositories/PromotionDialogRepositoryTests.cs

```diff
@@ -0,0 +1,107 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Modix.Data.Models.Promotions;
+using Modix.Data.Repositories;
+using Modix.Data.Test.TestData;
+using NSubstitute;
+using NUnit.Framework;
+using Shouldly;
+
+namespace Modix.Data.Test.Repositories
+{
+    public class PromotionDialogRepositoryTests
+    {
+        private static (ModixContext, PromotionDialogRepository) BuildTestContext()
+        {
+            var modixContext = TestDataContextFactory.BuildTestDataContext(x =>
+            {
+                x.Set<PromotionDialogEntity>().AddRange(PromotionDialogs.Entities.Clone());
+                x.Set<PromotionCampaignEntity>().AddRange(PromotionCampaigns.Entities.Clone());
+            });
+
+            var uut = new PromotionDialogRepository(modixContext);
+
+            return (modixContext, uut);
+        }
+
+        [Test]
+        public async Task GetAsync_DialogsExist()
+        {
+            var (modixContext, uut) = BuildTestContext();
+
+            var dialogs = await uut.GetDialogsAsync();
+
+            var dialogIds = PromotionDialogs.Entities
+                .Select(x => (x.MessageId, x.CampaignId))
+                .ToArray();
+
+            dialogs.Select(x => (x.MessageId, x.CampaignId))
+                .ShouldBe(dialogIds, true);
+        }
+
+        [TestCaseSource(nameof(NewDialogCreationTestCases))]
+        public async Task CreateAsync_InsertsDialog(PromotionDialogEntity data)
+        {
+            var (modixContext, uut) = BuildTestContext();
+
+            await uut.CreateAsync(data.MessageId, data.CampaignId);
+
+            modixContext.Set<PromotionDialogEntity>().ShouldContain(
+                x => x.CampaignId == data.CampaignId
+                     && x.MessageId == data.MessageId);
+            var user = modixContext.Set<PromotionDialogEntity>().First(x => x.CampaignId == data.CampaignId);
```

Comment:
`user` is kind of a confusing name. Maybe `newDialog` or something instead? Or could probably just remove this line entirely and reference `data.CampaignId` and `data.MessageId` below